lighthouse/constants/aggregation_stages.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any, Dict, Final, List
+
+from lighthouse.constants.fields import (
+    FIELD_FILTERED_POSITIVE,
+    FIELD_MUST_SEQUENCE,
+    FIELD_PREFERENTIALLY_SEQUENCE,
+    FIELD_PROCESSED,
+    FIELD_SAMPLE_ID,
+)
+
+"""
+Stage for mongo aggregation pipeline to select all the samples which are "fit to pick":
+- we first need to merge the fields from the priority_samples collection
+- we are then interested in samples which are:
+    filtered_positive == True OR must_sequence == True OR preferentially_sequence == True
+"""
+STAGES_FIT_TO_PICK_SAMPLES: Final[List[Dict[str, Any]]] = [
+    # first perform a lookup from samples to priority_samples using the '_id' field from samples on 'sample_id' on
+    #   priority_samples
+    {
+        "$lookup": {
+            "from": "priority_samples",
+            "let": {FIELD_SAMPLE_ID: "$_id"},
+            "pipeline": [
+                {
+                    "$match": {
+                        "$expr": {
+                            "$and": [
+                                {"$eq": [f"${FIELD_SAMPLE_ID}", f"$${FIELD_SAMPLE_ID}"]},
+                                {"$eq": [f"${FIELD_PROCESSED}", True]},
+                            ]
+                        }
+                    }
+                },
+                # include a project here to remove the other fields we are not interested in or could cause confusion
+                #   such as '_created_at' and '_updated_at' which are automatically created by Eve
+                {"$project": {FIELD_MUST_SEQUENCE: 1, FIELD_PREFERENTIALLY_SEQUENCE: 1}},
+            ],
+            "as": "from_priority_samples",
```

Comment:
It went a bit slow when we tried it in local from priority samples with samples. Joining was quicker with the other $lookup() with localField and foreignField.. and using a pre $match() for unprocessed, I think this one is for situations where you have to do conversions in the column values.